java/code/src/com/suse/manager/virtualization/VirtManagerSalt.java

```diff
@@ -166,6 +174,14 @@ public void updateLibvirtEngine(MinionServer minion) {
         pillar.put("virt_entitled", minion.hasVirtualizationEntitlement());
         saltApi.callSync(State.apply(Collections.singletonList("virt.engine-events"),
                 Optional.of(pillar)), minion.getMinionId());
+
+        if (minion.hasVirtualizationEntitlement()) {
+            minionVirtualizationPillarFileManager.generatePillarFile(minion);
+        }
+        else {
+            minionVirtualizationPillarFileManager.removePillarFile(minion.getMinionId());
+        }
+        SaltService.INSTANCE.refreshPillar(new MinionList(minion.getMinionId()));
```

Comment:
why is this needed?

Reply:
Without the `refreshPillar()` the pillar file is generated and the results can be seen in `salt 
pillar.items` but the change isn't reflected on the minion itself. Maybe there is a better place for this call, but without it the virtpoller is configured but not enabled on the minion

Reply:
I see

Reply:
As I understood all Salt calls should go through `SystemQuery` now,  so do we really want to make an exception for `refreshPillar()` and call it directly on the `SaltService` instance? Shouldn't we rather move `refreshPillar()` to the `SystemQuery` interface?

Reply:
Done!